### ISSUE TITLE
💅 noShadow false positive on constructor overload signatures

### ISSUE THREAD

**Issue by hyperknot** [biomejs/biome#8416]
### Environment information

```bash
playground
```

### Rule name

noShadow

### Playground link

https://next.biomejs.dev/playground/?lintRules=noShadow&code=YwBsAGEAcwBzACAARgBvAG8AIAB7AAoAIAAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoACkACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAeAA6ACAAcwB0AHIAaQBuAGcAKQAKACAAIABjAG8AbgBzAHQAcgB1AGMAdABvAHIAKAB4AD8AOgAgAHMAdAByAGkAbgBnACkAIAB7AH0ACgB9AA%3D%3D&ruleDomains.solid=recommended

### Expected result

No warning (overload signatures don't create variables)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Bertie690** [comment]
Duplicate of [#7125](https://github.com/biomejs/biome/issues/7125), for what it's worth.
